integration_tests/test_ibc.py

```diff
@@ -0,0 +1,188 @@
+    raw = next(iter(cluster.values())).raw
+    relayer = ["relayer", "--home", data_root / "relayer"]
+    # init light clients
+    for chain_id in cluster:
+        subprocess.run(
+            relayer
+            + [
+                "light",
+                "init",
+                chain_id,
+                "-f",
+            ],
+            check=True,
+        )
+
+    # setup path
+    subprocess.run(
+        relayer
+        + [
+            "tx",
+            "link",
+            "demo",
+            "-d",
+            "-o",
+            "3s",
+        ],
+        check=True,
+    )
+
+    # check status
+    rsp = json.loads(subprocess.check_output(relayer + ["chains", "list", "--json"]))
+    assert rsp == [
+        {
+            "key": "relayer",
+            "chain-id": "ibc-0",
+            "rpc-addr": "http://localhost:26657",
+            "account-prefix": "cro",
+            "gas-adjustment": 1.5,
+            "gas-prices": "0.0basecro",
+            "trusting-period": "336h",
+        },
+        {
+            "key": "relayer",
+            "chain-id": "ibc-1",
+            "rpc-addr": "http://localhost:26757",
+            "account-prefix": "cro",
+            "gas-adjustment": 1.5,
+            "gas-prices": "0.0basecro",
+            "trusting-period": "336h",
+        },
+    ]
+    rsp = json.loads(subprocess.check_output(relayer + ["paths", "list", "--json"]))
+    assert rsp == {
+        "demo": {
+            "src": {
+                "chain-id": "ibc-0",
+                "client-id": "demo0client",
+                "connection-id": "demo0connection",
+                "channel-id": "demo0channel",
+                "port-id": "transfer",
+                "order": "unordered",
+                "version": "ics20-1",
+            },
+            "dst": {
+                "chain-id": "ibc-1",
+                "client-id": "demo1client",
+                "connection-id": "demo1connection",
+                "channel-id": "demo1channel",
+                "port-id": "transfer",
+                "order": "unordered",
+                "version": "ics20-1",
+            },
+            "strategy": {
+                "type": "naive",
+            },
+        }
+    }
+    # query balance of relayer account
+    for chain_id in cluster:
+        assert (
+            subprocess.check_output(relayer + ["query", "balance", chain_id])
+            .strip()
+            .decode()
+        ) == "10000000000basecro"
+
+    # do a transfer from ibc-0 to ibc-1
+    recipient = cluster["ibc-1"].address("relayer")
+    subprocess.run(
+        relayer + ["tx", "transfer", "ibc-0", "ibc-1", "10000basecro", recipient],
+        check=True,
+    )
+    # sender balance decreased
+    assert cluster["ibc-0"].balance(cluster["ibc-0"].address("relayer")) == 9999990000
+
+    subprocess.run(relayer + ["tx", "relay", "demo", "-d"], check=True)
+
+    denom_hash = hashlib.sha256(b"transfer/demo1channel/basecro").hexdigest().upper()
```

Comment:
for the denominations, isn't there some standard convention (to also include chain-id and should chain-id have some format)?

Reply:
this format is documented here: https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-001-coin-source-tracing.md#decision